catalyst-gateway/bin/src/service/common/types/generic/ed25519_public_key.rs

```diff
@@ -133,3 +133,24 @@ impl From<Ed25519HexEncodedPublicKey> for ed25519_dalek::VerifyingKey {
             .expect("This can only fail if the type was invalidly constructed.")
     }
 }
+
+impl TryInto<Vec<u8>> for Ed25519HexEncodedPublicKey {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<Vec<u8>, Self::Error> {
+        Ok(hex::decode(self.0.trim_start_matches("0x"))?)
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Could also add some comments from where you got these test vectors.

Reply:
added[ this](https://cexplorer.io/article/understanding-cardano-addresses) as its the best resource